Fix CURLOPT_POSTFIELDS param type

ahmedash95 opened the pull request:
Hi Ondrej,

`CURLOPT_POSTFIELDS` accepts string or array but PHPStan reports a type error if the value is not a string
https://phpstan.org/r/db599e65-dfae-4e85-90bc-6bc96feadf96

I'm not sure about `TypeCombinator::union` but it's what I was able to find to achieve the fix. 

closes https://github.com/phpstan/phpstan/issues/8053.

Thanks.

ahmedash95 committed:
Fix CURLOPT_POSTFIELDS param type

ondrejmirtes commented:
Hi Ahmed :) There's already https://github.com/phpstan/phpstan-src/pull/1777 but I can't merge it because of coding standards and requirements checker failures. So yours PR might be correct sooner and might win the race :)

Your PR passes but is missing any tests - it'd be nice to add tests for cases when it should pass, and at least one test for the case when the rule should report an error.

Also please base the PR on 1.8.x so it can be released sooner.

ahmedash95 committed:
adding tests

ahmedash95 commented:
Thanks for pointing out the other PR. I did not know about it because it's not mentioned in the issue.

I've added the tests and changed that base 1.8.x. I assume it's ready to be merged in case the other one is not.

ondrejmirtes commented:
Thank you!
